### ISSUE TITLE
TinyMCE - Configure Plugin

### ISSUE THREAD

**Issue by katknow** [omeka/Omeka#1144]
# Configure Plugin - '...' toolbar menu button items are not announced (VoiceOver) (3rd Party - TinyMCE)

"Description: On the 'Configure Plugin' page, users are able to interact with a '...' more options button to expand a submenu. However, with the VoiceOver screen reader on Safari, the components within the expanded menu are not announced. These include the 'Insert/edit link' button, the 'Blocks' combobox, and the 'Source Code' button.

Note: This also occurs under the HTML toolbars of the add items page.

Note 2 : This behavior is also observed on the Appearance Themes → Configure Theme page, where the components within the ‘…’ more options menu similarly fail to announce their accessible names, roles, or values when using VoiceOver on Safari.

Note: The Text Styles combobox items are intermittently announced and they are frequently not announced on MacOS 26.1.

Expected Results: All components announce their accessible name, their role, and their current state/value as needed. 

We are not sure why the Text Styles combobox is not announcing list items correctly, but the Select Only Combobox example can be used as a comparison. https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/

Actual Results: The components within the '...' more options submenu are not announced when they receive focus with the VoiceOver screenreader. "

4.1.2 Name, Role, Value; Level A; WCAG 2.0
1.3.1 Info and Relationships; Level A; WCAG 2.0

# Configure Plugin - '...' toolbar sub-menu does not trap focus (VoiceOver) (3rd Party - TinyMCE)

"Description: On the 'Configure Plugin' page, users are able to interact with a '...' more options button to expand a submenu. However, with the VoiceOver screen reader on Safari, focus is able to escape the expanded menu. Once focus has escaped, focus is unable to reenter the submenu. To work around this, users must interact with the '...' button again to collapse the menu and then expand it again. 

Note: This also occurs in the HTML options of the add items page. When the checkbox is selected to use HTML, the toolbar appears.

Expected Results: When a menu button has been expanded to reveal a submenu, focus is trapped within the menu until users make a selection or dismiss the menu by pressing the Esc key.

Actual Results: With VoiceOver on Safari browser, focus is able to escape the '...' more options expanded submenu. Once focus has escaped, it cannot reenter the menu."

2.4.3 Focus Order; Level A; WCAG 2.0

# Configure Plugin - 'Text Styles' combobox lacks persistent visual label (3rd Party - TinyMCE)
"Description: On the 'Configure Plugin' page, users are able to interact with a '...' more options button. Within the expanded submenu is a 'Blocks' menu button to allow users to select a text style. Currently, this menu button/combobox relies on the current selection as it's visible label and lacks a persistent visible label identifying what the component is for. 

Note: This same issue is also present on the Appearance Themes uder Configure Theme section. The UI pattern for the style/format selection combobox also lacks a persistent visible label and relies solely on the current selection as the displayed label.

Expected Results: The 'Blocks' menu button/combobox has a persistent visible label that matches the component's accessible name. 

Actual Results: The 'Blocks' menu button/combobox lacks a persistent visible label and relies on the current selection as a label. "

3.3.2 Labels or Instructions; Level A; WCAG 2.0

# Configure Plugin - Focus can escape modals (3rd Party - TinyMCE)

"Description: On the 'Configure Plugin' page, users are able to open a 'Source Code' or 'Insert/Edit Link' modal within the multi-line text editor toolbars. The buttons to open these modals are found within the '...' more options button. 

It was found that when using the Up Arrow key to encounter the visual heading of 'Insert/Edit Link' or 'Source Code' and then pressing the Tab key to navigate back to  an interactable component, focus will move to the 'Close' button (Which is normally not in tab order as described in bug 48). 

Pressing Tab again will cause focus to leave the modal entirely as focus moves to the 'Skip to Main Page' link. Continuing to press the Tab key to navigate, focus will encounter all interactable components on the main page while the modal remains open. It was found that the modal cannot be reencountered with Tab key once focus has escaped this way, resulting in it being stuck open on the screen blocking content. The modal can be reencountered with Arrow keys.

Note: Could not reproduce on MacOS with Safari

Note: This was also found to occur on the 'Add an Item' page under the 'Dublin Core' and 'Item Type Metadata' tabs.

Expected Results: Focus is unable to escape a modal until users close/dismiss it or complete the task within it. 

Actual Results: Focus is able to escape the 'Source Code' and 'Insert/Edit Link' modals. Once focus escapes, the modal can no longer be encountered with Tab navigation. The modal remains open blocking main page content. "

2.4.3 Focus Order; Level A; WCAG 2.0
2.4.11 Focus Not Obscured (Minimum); Level AA; WCAG 2.2

**Comment by kimisgold** [comment]
Could we get this retested? I think these have all been addressed to some degree in tinymce 5.

**Comment by aviannamiller** [comment]
Will retest, but first I'm noting that the buttons in the toolbar are preventing keyboard access as they include `tabindex="-1"`. This failure to navigate the toolbar seems to have already been pointed out on their repo, but there's no apparent progress. We could add to the reporting there if that is the best approach.

**Comment by kimisgold** [comment]
Hmm, I recall they have [their own keyboard shortcuts](https://www.tiny.cloud/docs/tinymce/5/keyboard-shortcuts/) for navigating the toolbar content. Do those work, and would that satisfy the keyboard access requirement?
